Directory.Packages.props

```diff
@@ -0,0 +1,141 @@
+    <PackageVersion Include="System.Security.Cryptography.Pkcs" Version="$(CryptographyPackagesVersion)" />
+    <PackageVersion Include="System.Security.Cryptography.ProtectedData" Version="4.4.0" />
+    <PackageVersion Include="System.Text.Encoding.Extensions" Version="$(SystemPackagesVersion)" />
+    <PackageVersion Include="System.Threading" Version="$(SystemPackagesVersion)" />
+    <PackageVersion Include="System.Threading.Tasks" Version="$(SystemPackagesVersion)" />
+    <!--
+      The Microsoft.VisualStudio.SDK metapackage brings in System.Threading.Tasks.Dataflow 4.11.1 (assembly version 4.9.5.0).
+      However, our MSBuild integration tests use Microsoft.Build 16.8.0, which requires System.Threading.Tasks.Dataflow 4.9.0 (assembly version 4.9.3.0).
+      To resolve runtime assembly binding failures, we'll downgrade the package from 4.11.1 to 4.9.0.
+    -->
+    <PackageVersion Include="System.Threading.Tasks.Dataflow" Version="4.9.0" />
+    <PackageVersion Include="VSSDK.TemplateWizardInterface" Version="12.0.4" />
+    <PackageVersion Include="VsWebSite.Interop" Version="$(VSFrameworkVersion)" />
+    <PackageVersion Include="xunit" Version="2.4.1" />
+    <PackageVersion Include="xunit.runner.visualstudio" Version="2.4.5" />
+    <PackageVersion Include="Xunit.StaFact" Version="1.0.37" />
+  </ItemGroup>
+
+  <!--
+    Packages that provide NuGet's Visual Studio extensibility APIs should not depend on the Visual Studio SDK, to ensure
+    there are no circular references in case the VS SDK itself adds our package as a dependency. Everything else, however
+    should not add PackageReference on any package that is part of the VS SDK.
+
+    These individual package versions should be equal to or less than the package version within the VS SDK that our
+    customers use, otherwise they might get NU1605 or MSB3277 warnings.
+  -->
+  <ItemGroup Condition=" '$(MSBuildProjectFile)' == 'NuGet.VisualStudio.csproj' OR
+              '$(MSBuildProjectFile)' == 'NuGet.VisualStudio.Contracts.csproj' OR
+              '$(MSBuildProjectFile)' == 'NuGet.VisualStudio.Interop.csproj' OR
+              '$(MSBuildProjectFile)' == 'NuGet.SolutionRestoreManager.Interop.csproj'">
+    <PackageVersion Include="Microsoft.ServiceHub.Framework" Version="2.7.327-preview" />
+    <PackageVersion Include="Microsoft.VisualStudio.ComponentModelHost" Version="17.0.67-g9e37b637e6" />
+    <PackageVersion Update="Microsoft.VisualStudio.SDK" Version="" />
+    <PackageVersion Include="Microsoft.VisualStudio.Shell.15.0" Version="$(VSFrameworkVersion)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <GlobalPackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" Condition="'$(Shipping)' == 'true' AND '$(IsXPlat)' != 'true'" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(UsePublicApiAnalyzer)' == 'true' ">
+    <GlobalPackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="$(MicrosoftCodeAnalysisPublicApiAnalyzersVersion)" />
```

Comment:
We probably need to wait to merge this until we know 100% all the CIs have 17.4. 

We should also update our requirements section for building the repo.